lib/sycamore/sycamore/document.py

```diff
@@ -0,0 +1,388 @@
+        self._children = children
+
+    def children(self) -> list[Node]:
+        return self._children
+
+
+class Caption(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_caption(self)
+
+
+class Footnote(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_footnote(self)
+
+
+class Formula(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_formula(self)
+
+
+class ListItem(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_list_item(self)
+
+
+class PageFooter(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_page_footer(self)
+
+
+class PageHeader(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_page_header(self)
+
+
+class Picture(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_picture(self)
+
+
+class SectionHeader(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_section_header(self)
+
+
+class Table(Leaf):
+    def __init__(self, node_id, box, content, metadata, continued=None):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_table(self)
+
+
+class Text(Leaf):
+    def __init__(self, node_id, box, content, metadata, continued=None):
+        super().__init__(node_id, box, content, metadata)
+        self._continued = continued
+
+    def category(self):
+        return Category.Caption
+
+    def continued(self):
```

Comment:
What are the semantics here? Why does Text have a continued field, but not, say, tables?

Reply:
This is for the split objects. Table does have, it's just nothing is support there like merge across pages.